python/audit/utils.py

```diff
@@ -253,25 +252,26 @@ def audit_finish_task(obj, action_display_name: str, status: str) -> None:
     if not obj_type:
         return
 
-    if obj_type == AuditObjectType.Host:
-        obj_name = obj.fqdn
-    else:
-        obj_name = obj.name
-
     audit_object = get_or_create_audit_obj(
         object_id=obj.pk,
-        object_name=obj_name,
+        object_name=obj.name,
         object_type=obj_type,
     )
     if status == "success":
         operation_result = AuditLogOperationResult.Success
     else:
         operation_result = AuditLogOperationResult.Fail
 
+    user = None
+    audit_log = AuditLog.objects.filter(audit_object=audit_object).first()
```

Comment:
This request does not guarantee the determination of the correct user. We want the correct user to be the one who started the task.

Reply:
Refactored